app/src/main/java/com/github/swent/echo/compose/map/MapLibreMapViewProvider.kt

```diff
@@ -79,36 +88,37 @@ class MapLibreMapViewProvider : IMapViewProvider<MapView> {
         val styleUrl =
             context.getString(R.string.maptiler_base_style_url) +
                 context.getString(R.string.maptiler_api_key)
-        val isInit = (mapView == null)
-        if (isInit) {
-            // Creating the map for the first time
-            mapView = MapView(context)
-        }
-        mapView?.onCreate(null)
-        mapView?.getMapAsync { map ->
+
+        val mapView = MapView(context)
+        mapView.onCreate(null)
+        mapView.getMapAsync { map ->
             // Set the style after mapView was loaded
             map.setStyle(styleUrl) {
                 map.uiSettings.apply {
                     setAttributionMargins(15, 0, 0, 15)
                     isRotateGesturesEnabled = true
                 }
-                // Set the map view center if we're creating the view
-                if (isInit) {
-                    map.cameraPosition =
-                        CameraPosition.Builder()
-                            .target(MAP_CENTER.toLatLng())
-                            .zoom(DEFAULT_ZOOM)
-                            .bearing(0.0)
-                            .build()
-                }
+
+                // Restore the last camera position
+                map.cameraPosition = lastCameraPosition
+
+                // Add listener to handle long press events
                 map.addOnMapLongClickListener {
                     onLongPress(it)
                     true
                 }
+
+                // Add listener to update the last camera position so we can restore it later
+                map.addOnCameraMoveListener {
+                    lastCameraPosition = CameraPosition.Builder(map.cameraPosition).build()
```

Comment:
I believe it could be more readable (and optimal) to assign the map's camera position directly instead of building a new one each time the map moves.